pilot/pkg/security/model/authentication.go

```diff
@@ -217,10 +228,15 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, metadata *model.N
 			CaCertificatePath: metadata.TLSServerRootCert,
 		}
 
+		matchSAN := util.StringToExactMatch(subjectAltNames)
```

Comment:
TODO: if subjectAltName ends with *, create a prefix match as well. 
TODO: if user explicitly specifies SANs - should we alter his explicit config by adding all spifee aliases ? I think we need to be very careful with that - it doesn't seem we have any way to disable adding the spiffee aliases. Maybe until we have we should just use them as default ( i.e. if SAN list is set - use it, user will put explicily what spifee URL prefixes they want)

Reply:
And the real reason for that: user can specify for example spiffe://TD1/namespace/NS1/sa/SA! - to indicate explicitly that only SA1 in NS1 in TD1 are allowed. This is pretty clearly not intending to allow EVERYONE in all TD, NS and SAs in the mesh. 

 I think we should add the equivalent Gateway setting for Sidecar - once again the lack of consistency between the 2 modes is hurting us ( luckily K8s Gateway covers sidecars as well )

Reply:
> Re Gateway: we can add a second flag to disable this for Gateaway only if we think Shriram's use case is valid. AFAIK we don't officially or unofficially support a mesh with multiple trust domains - except via the 'aliases' feature. The use case doesn't make sense to me - how do we know that the added clusters are actually your clusters and not some attacker cluster ? If the user doesn't explicitly specify which are his clusters - there is no way to know.

cc: @rshriram. Also I don't think adding one more flag just for gateway is good idea. But I do agree that we should ideally not have inconsistencies between sidecar and gateway

Reply:
I will add the following TODO's to this. But should this block us from merging this PR? With 1.7 release in 5 days, I think we do want to ship it out ASAP.